ambry-store/src/main/java/com.github.ambry.store/StoreMetrics.java

```diff
@@ -143,7 +143,13 @@ public Long getValue() {
         return index.hardDeleteCaughtUp() ? 1L : 0L;
       }
     };
-    registry.register(MetricRegistry.name(PersistentIndex.class, name + "-hardDeleteCaughtUp"),
-        hardDeleteCaughtUp);
+    registry.register(MetricRegistry.name(PersistentIndex.class, name + ".HardDeleteCaughtUp"), hardDeleteCaughtUp);
+  }
+
+  static String getPartitionName(String replicaDir) {
```

Comment:
+1. We should just pass in the final name to the `StoreMetrics` constructor. Any processing you have to do on the name should be outside of `StoreMetrics`.